uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -1820,6 +1820,108 @@ Feature: GGMQ-1
+  Scenario Outline: GGMQ-1-T25-<mqtt-v>-<name>: As a customer, I connect with same client id twice (not reconnect) and be able to publish and subscribe to messages.
+    When I create a Greengrass deployment with components
+      | aws.greengrass.clientdevices.Auth        | LATEST                                  |
+      | aws.greengrass.clientdevices.mqtt.EMQX   | LATEST                                  |
+      | aws.greengrass.clientdevices.IPDetector  | LATEST                                  |
+      | <agent>                                  | classpath:/local-store/recipes/<recipe> |
+    And I create client device "publisher"
+    And I create client device "subscriber"
+    When I associate "publisher" with ggc
+    When I associate "subscriber" with ggc
+    And I update my Greengrass deployment configuration, setting the component aws.greengrass.clientdevices.Auth configuration to:
+    """
+{
+    "MERGE":{
+        "deviceGroups":{
+            "formatVersion":"2021-03-05",
+            "definitions":{
+                "MyPermissiveDeviceGroup":{
+                    "selectionRule":"thingName: ${publisher} OR thingName: ${subscriber}",
+                    "policyName":"MyPermissivePolicy"
+                }
+            },
+            "policies":{
+                "MyPermissivePolicy":{
+                    "AllowAll":{
+                        "statementDescription":"Allow client devices to perform all actions.",
+                        "operations":[
+                            "*"
+                        ],
+                        "resources":[
+                            "*"
+                        ]
+                    }
+                }
+            }
+        }
+    }
+}
+    """
+
+    And I update my Greengrass deployment configuration, setting the component <agent> configuration to:
+    """
+{
+    "MERGE":{
+        "controlAddresses":"${mqttControlAddresses}",
+        "controlPort":"${mqttControlPort}"
+    }
+}
+    """
+
+    And I deploy the Greengrass deployment configuration
+    Then the Greengrass deployment is COMPLETED on the device after 5 minutes
+    And the aws.greengrass.clientdevices.mqtt.EMQX log on the device contains the line "is running now!." within 1 minutes
+
+    Then I discover core device broker as "localMqttBroker1" from "publisher" in OTF
+    Then I discover core device broker as "localMqttBroker2" from "subscriber" in OTF
+    And I connect device "publisher" on <agent> to "localMqttBroker1" using mqtt "<mqtt-v>"
+    And I connect device "subscriber" on <agent> to "localMqttBroker2" using mqtt "<mqtt-v>"
+
+    When I subscribe "subscriber" to "iot_data_0" with qos 1
+    When I publish from "publisher" to "iot_data_0" with qos 1 and message "Test message"
+    And message "Test message" received on "subscriber" from "iot_data_0" topic within 5 seconds
+    And I rename connection "publisher" to "publisher_old"
+
+    # Reconnect publisher with the same device id
```

Comment:
How can we check that the old client loses the connection due to duplicate client ID?

Reply:
How do you know if this works? Do you have logs showing that the duplicate client was disconnected?

Reply:
In that T25 we just check second connection is alive and we can publishing messages over it.
It repeat the same steps as original GGAD-1-T25.
We expecting add steps to checking disconnection when implement T26 scenario.
In addition when T26 will ready we can update T25 to add step with check connection has been really disconnected.
We expecting T26 takes a time to implement that check.

Reply:
How do you know that this actually creates a new connection and doesn't just reuse the existing connection? Can you show me a log showing the disconnect and new connect call?

Reply:
I've attached additional logs, where we can see that broker disconnects the previous connection
In client log:
```
2023-08-01T17:42:25.260Z [INFO] (Copier) aws.greengrass.client.Mqtt5JavaPahoClient: stdout. [INFO ] 2023-08-01 20:42:25.261 [grpc-default-executor-0] GRPCControlServer - createMqttConnection: clientId gg-df6909aca85d4c94507b-publisher broker 192.168.100.20:8883. {scriptName=services.aws.greengrass.client.Mqtt5JavaPahoClient.lifecycle.Run, serviceName=aws.greengrass.client.Mqtt5JavaPahoClient, currentState=RUNNING}
2023-08-01T17:42:26.468Z [INFO] (Copier) aws.greengrass.client.Mqtt5JavaPahoClient: stdout. [INFO ] 2023-08-01 20:42:26.467 [MQTT Rec: gg-df6909aca85d4c94507b-publisher] MqttConnectionImpl - MQTT connectionId 1 disconnected error 'null' disconnectInfo 'com.aws.greengrass.testing.mqtt5.client.GRPCClient$DisconnectInfo@42609add'. {scriptName=services.aws.greengrass.client.Mqtt5JavaPahoClient.lifecycle.Run, serviceName=aws.greengrass.client.Mqtt5JavaPahoClient, currentState=RUNNING}
2023-08-01T17:42:26.477Z [INFO] (Copier) aws.greengrass.client.Mqtt5JavaPahoClient: stdout. [INFO ] 2023-08-01 20:42:26.476 [grpc-default-executor-0] GRPCControlServer - Publish: connectionId 3 topic 'iot_data_0' QoS 1 retain false. {scriptName=services.aws.greengrass.client.Mqtt5JavaPahoClient.lifecycle.Run, serviceName=aws.greengrass.client.Mqtt5JavaPahoClient, currentState=RUNNING}
2023-08-01T17:42:26.486Z [INFO] (Copier) aws.greengrass.client.Mqtt5JavaPahoClient: stdout. [INFO ] 2023-08-01 20:42:26.483 [grpc-default-executor-0] GRPCControlServer - Publish response: connectionId 3 reason code 0 reason string ''. {scriptName=services.aws.greengrass.client.Mqtt5JavaPahoClient.lifecycle.Run, serviceName=aws.greengrass.client.Mqtt5JavaPahoClient, currentState=RUNNING}

```
In test console log:
```
[INFO ] 2023-08-01 20:42:26.581 [grpc-default-executor-0] GRPCDiscoveryServer - OnMqttDisconnect: agentId aws.greengrass.client.Mqtt5JavaPahoClient connectionId 1 disconnect 'reasonCode: 142
' error ''
[INFO ] 2023-08-01 20:42:26.584 [grpc-default-executor-0] MqttControlSteps - MQTT client disconnected. Error: 
```